test/e2e_node/device_plugin_test.go

```diff
@@ -687,21 +687,21 @@ func testDevicePlugin(f *framework.Framework, pluginSockDir string) {
 			gomega.Expect(resourcesForOurPod.Name).To(gomega.Equal(pod1.Name))
 			gomega.Expect(resourcesForOurPod.Namespace).To(gomega.Equal(pod1.Namespace))
 
-			// Note that the kubelet does not report resources for restartable
-			// init containers for now.
-			// See https://github.com/kubernetes/kubernetes/issues/120501.
-			// TODO: Fix this test to check the resources allocated to the
-			// restartable init container once the kubelet reports resources
-			// for restartable init containers.
```

Comment:
https://github.com/kubernetes/kubernetes/pull/120461#discussion_r1317515863

removed this TODO and fixed the e2e test to check both restartable init container and regular container's device information as this PR is trying to address the #120501.
@ffromani